### PR TITLE
(GH-1057) Regex fix to allow dotted resources

### PR DESCRIPTION
**Pull request by LukasAud**
Prior to this commit, one of our recent module updates introduced a
regex validation step for the resource names in our ppa.pp manifest
which would raise an issue if a valid resource name contained a dot (.).

This commit aims to slightly adjust the regex validation so that it
allows for dotted resource names. This PR should fix issue #1057.

### CODE REVIEW

**Review by chelnak — CHANGES_REQUESTED** [review]
Please can you update the unit tests to reflect this change?

File: spec/defines/ppa_spec.rb [hunk 1]
```diff
@@ -43,6 +43,68 @@ def ppa_exec_params(user, repo, distro = 'trusty', environment = [])
...
+  end
+
+  [
+    'ppa:foo!/bar',
+    'ppa:foo/bar!',
+    'ppa:foo1.0/bar',
+    'ppa:foo/bar/foobar',
+    '|| ls -la ||',
+    '|| touch /tmp/foo.txt ||',
+  ].each do |inv_value|
```
Comment (chelnak) [inline spec/defines/ppa_spec.rb:83]: This doesn't need to be different as it is scoped locally to the each. We can just keep it as `value` for simplicity. 

```suggestion
  ].each do |value|
```

File: spec/defines/ppa_spec.rb [hunk 2]
```diff
@@ -43,6 +43,68 @@ def ppa_exec_params(user, repo, distro = 'trusty', environment = [])
...
+            },
+            distro: {
+              codename: 'trusty',
+              id: 'Ubuntu',
+            },
+          },
+        }
+      end
+
+      let(:title) { inv_value }
```
Comment (chelnak) [inline spec/defines/ppa_spec.rb:102]: This doesn't need to be different as it is scoped locally to the each. We can just keep it as `value` for simplicity. 

```suggestion
      let(:title) { value }
```

File: spec/defines/ppa_spec.rb [hunk 3]
```diff
@@ -43,6 +43,68 @@ def ppa_exec_params(user, repo, distro = 'trusty', environment = [])
...
+
+      it { is_expected.to raise_error }
```
Comment (chelnak) [inline spec/defines/ppa_spec.rb:104]: Let's be more explicit about the error we are looking for here.

You can use [this](https://github.com/puppetlabs/puppetlabs-apt/blob/65e03244552e8113b69f6ca5f4e6dc142251df0b/spec/defines/mark_spec.rb#L92) as an example.

File: spec/defines/ppa_spec.rb [hunk 4]
```diff
@@ -43,6 +43,68 @@ def ppa_exec_params(user, repo, distro = 'trusty', environment = [])
...
+              codename: 'trusty',
+              id: 'Ubuntu',
+            },
+          },
+        }
+      end
+
+      let(:title) { value }
+
+      it { is_expected.not_to raise_error }
```
Comment (chelnak) [inline spec/defines/ppa_spec.rb:72]: For the happy path, we could also check if the `exec` is being called with the resource title that we expect.

There are a few examples in this spec that use the `contains_exec` method.

**Review by chelnak — CHANGES_REQUESTED** [review]
Almost there! It's looking great, just a couple more observations.

**Review by chelnak — APPROVED** [review]
Fantastic thanks for putting in the effort!

**Comment by pmcmaw** [comment]
Fixes #1057

**Comment by pmcmaw** [comment]
Fixed #1057

**Pull request merged**